packages-exp/auth-exp/src/core/persistence/persistence_user_manager.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Persistence } from '../persistence';
+import { User } from '../../model/user';
+import { ApiKey, AppName, Auth } from '../../model/auth';
+import { inMemoryPersistence } from './in_memory';
+import { UserImpl } from '../user/user_impl';
+
+export const AUTH_USER_KEY_NAME_ = 'authUser';
+export const PERSISTENCE_KEY_NAME_ = 'persistence';
+const PERSISTENCE_NAMESPACE_ = 'firebase';
+
+export function persistenceKeyName_(
+  key: string,
+  apiKey: ApiKey,
+  appName: AppName
+): string {
+  return `${PERSISTENCE_NAMESPACE_}:${key}:${apiKey}:${appName}`;
+}
+
+export class PersistenceUserManager {
+  private constructor(
+    public persistence: Persistence,
+    private readonly auth: Auth,
+    private readonly userKey: string
+  ) {}
+
+  fullKeyName_(key: string): string {
+    const { config, name } = this.auth;
+    return persistenceKeyName_(key, config.apiKey, name);
+  }
+
+  setCurrentUser(user: User): Promise<void> {
+    return this.persistence.set(this.fullKeyName_(this.userKey), user);
+  }
+
+  getCurrentUser(): Promise<User | null> {
+    return this.persistence.get<User>(
+      this.fullKeyName_(this.userKey),
+      (o: { [key: string]: unknown }) => UserImpl.fromPlainObject(this.auth, o)
```

Comment:
what's `o`?

Reply:
the blob. Anyway this is an instantiator. should be clear now